orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTask.java

```diff
@@ -83,6 +88,18 @@ public TaskResult execute(Stage stage) {
       }
 
       Application application = front50Service.get(applicationName);
+      List<String> processedInterestingHealthProviderNames = interestingHealthProviderNamesProcessors
+        .stream()
+        .filter(healthProcessor -> healthProcessor.supports(getCloudProvider(stage), application, stage))
+        .flatMap(healthProcessor -> healthProcessor.process(stage).stream())
+        .collect(Collectors.toList());
+
+      if (!processedInterestingHealthProviderNames.isEmpty()) {
```

Comment:
IMO we should only do this if we didn't find a specific processor ... as it's "possible" that a process might happen to return an empty list.